saleor/graphql/checkout/mutations.py

```diff
@@ -1032,6 +1046,139 @@ def perform_mutation(
         return CheckoutShippingMethodUpdate(checkout=checkout)
 
 
+class CheckoutDeliveryMethodUpdate(BaseMutation):
+    checkout = graphene.Field(Checkout, description="An updated checkout")
+
+    class Arguments:
+        token = UUID(description="Checkout token.", required=False)
+        shipping_method_id = graphene.ID(description="Shipping Method.")
+        collection_point_id = graphene.ID(description="Collection Point Id")
+
+    class Meta:
+        description = (
+            "Updates the delivery method (shipping method or pick up point)"
+            "of the checkout."
+        )
+        error_type_class = CheckoutError
+        error_type_field = "checkout_error"
+
+    @classmethod
+    def perform_on_shipping_method(
+        cls, info, shipping_method_id, checkout_info, lines, checkout, manager
+    ):
+        shipping_method = cls.get_node_or_error(
+            info,
+            shipping_method_id,
+            only_type=ShippingMethod,
+            field="shipping_method_id",
+            qs=shipping_models.ShippingMethod.objects.prefetch_related(
+                "postal_code_rules"
+            ),
+        )
+
+        delivery_method_is_valid = clean_delivery_method(
+            checkout_info=checkout_info,
+            lines=lines,
+            method=shipping_method,
+        )
+        if not delivery_method_is_valid:
+            raise ValidationError(
+                {
+                    "delivery_method": ValidationError(
+                        "This shipping method is not applicable.",
+                        code=CheckoutErrorCode.SHIPPING_METHOD_NOT_APPLICABLE,
+                    )
+                }
+            )
+
+        checkout.shipping_method = shipping_method
+        checkout.collection_point = None
+        checkout.save(
+            update_fields=["shipping_method", "collection_point", "last_change"]
+        )
+        recalculate_checkout_discount(
+            manager, checkout_info, lines, info.context.discounts
+        )
+        manager.checkout_updated(checkout)
+        return CheckoutDeliveryMethodUpdate(checkout=checkout)
+
+    @classmethod
+    def perform_on_collection_point(
```

Comment:
`perform_on_shipping_method` and `perform_on_collection_point` are very similar. I don't see a lot of differences - Different params for `cls.get_node_or_error`, different error msg and different assigment to checkout. IMO we could try to extract common part of this methods as most of the logic is duplicated.

Reply:
Done, I have extracted 2 methods.